app/models/course/material/folder.rb

```diff
@@ -1,7 +1,9 @@
 class Course::Material::Folder < ActiveRecord::Base
   acts_as_forest order: 'name'
+  include Course::ModelComponentHost::Component
 
   after_initialize :set_defaults, if: :new_record?
+  before_destroy :destroy_children
```

Comment:
@lowjoel Unfortunately, edge does not support dependent options :(
https://github.com/jackc/edge/blob/master/lib/edge/forest.rb#L10